app/src/androidTest/java/org/mozilla/focus/helpers/MockLocationUpdatesRule.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.helpers
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.SystemClock
+import android.util.Log
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import org.junit.rules.ExternalResource
+import org.mozilla.focus.helpers.TestHelper.executeShellCommandBlocking
+import java.util.Date
+import kotlin.random.Random
+
+private const val mockProviderName = LocationManager.GPS_PROVIDER
+
+/**
+ * Rule that sets up a mock location provider that can inject location samples
+ * straight to the device that the test is running on.
+ */
+class MockLocationUpdatesRule : ExternalResource() {
```

Comment:
If this is inspired from other project we should add a comment similar to https://github.com/mozilla-mobile/android-components/blob/1738f0558082eebd38bc1e705fdf22012dbd6e3f/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlinx/coroutines/Utils.kt#L18 to give credit to the original project.

Reply:
done. thanks Petru!